sample_app/app/models/user.rb

```diff
@@ -0,0 +1,14 @@
+class User < ApplicationRecord
+  before_save :downcase_email
+  validates :name, presence: true, length: {maximum: 50}
+  VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-]+(\.[a-z\d\-]+)*\.[a-z]+\z/i
+  validates :email, presence: true, length: {maximum: 255},
+    format: {with: VALID_EMAIL_REGEX},
+    uniqueness: {case_sensitive: false}
+  has_secure_password
```

Comment:
phần trên là validate cho trường email ( bắt buộc, độ dài, format, và k trùng lặp) ạ, còn dòng cuối là thêm vào để giúp hash password để lưu vào database thay vì thêm thẳng password thật => an toàn hơn ạ